cmd/dagger/module_inspect.go

```diff
@@ -97,28 +99,29 @@ func tryInitializeModule(ctx context.Context, dag *dagger.Client, srcRef string)
 	defer telemetry.End(span, func() error { return rerr })
 
 	findCtx, findSpan := Tracer().Start(ctx, "finding module configuration", telemetry.Encapsulate())
-	conf, _ := getModuleConfigurationForSourceRef(findCtx, dag, srcRef, true, true)
+	modSrc := dag.ModuleSource(srcRef)
```

Comment:
We need `AllowNotExists` here:
```suggestion
	modSrc := dag.ModuleSource(srcRef, dagger.ModuleSourceOpts{AllowNotExists: true})
```

Reply:
Why would we want AllowNotExists? The intention for allow not exists was really just to support `dagger init` when it's possible the source root dir doesn't exist yet locally. When would `dagger shell` or `dagger call` want to load a module that doesn't exist?

There's complication with enabling it because it in combination with find-up means you can specify any subdir that doesn't exist and just "find-up" to a module that does exist. So like `dagger -m this/is/not/real ...` in the dagger repo results in the `dagger-dev` module being loaded

Reply:
> Why would we want AllowNotExists? The intention for allow not exists was really just to support `dagger init` when it's possible the source root dir doesn't exist yet locally. 

I get that, but there’s two other use cases that I can think of. 

> When would `dagger shell` or `dagger call` want to load a module that doesn't exist?

It's not that the shell needs to load a module that doesn't exist, it just needs to not fail if the given ref isn't a module. The reason for that is that the shell supports looking into multiple places when you call a function.

For example:
```
⋈ alpine | .doc
⋈ github.com/dagger/dagger/modules/alpine | .doc
```
1. Is it a function in currently loaded module?
2. Is it a dependency of the currently loaded module?
3. Is it a stdlib function?
4. Is it a reference to a module? (this is where I use `tryInitializeModule`)
5. If none of the above, return simple error

This is the same kind of behavior on `dagger query` and `dagger listen`, which load a module optionally if it exists in the current dir. `dagger call` doesn’t need this because it has to be started with a valid module. `dagger shell` doesn’t.

The motivation was mainly to depend on `ConfigExists` rather than ignoring an error:
```
⋈ .core | module-source github.com/dagger/dagger-go-sdk/telemetry | config-exists
Error: input: moduleSource git module source "github.com/dagger/dagger-go-sdk/telemetry" does not contain a dagger config file
```

Because ignoring the error could hide other things that went wrong, not necessarily that there's no module. Besides, it prevents a failing span to show up in telemetry.

But things are a bit different now, and since the _trying_ to load a module is the last step in that fallback strategy, I can just let it return what `ModuleSource` returns, failure or not. But I still need to ignore the error when the shell is started, just like in `optionalModCmdWrapper`.

That being said, even if we end up wanting to make check without an error later on, it can be triggered by a separate option in `ModuleSource` or a separate field altogether, which don't break backwards compat since it would be an addition.